### PR TITLE
docs: Update and add units to the docstrings

### PR DESCRIPTION
**Pull request by uwcdc**
**Description:**
This pull request aims to enhance the documentation by updating existing docstrings and adding units to clarify parameter values. The goal is to improve readability and provide more comprehensive information to users and developers.

**Changes Made:**
- Updated existing docstrings across the codebase to ensure accuracy and clarity.
- Added units to parameter and return descriptions in the docstrings where applicable, providing context for the values.
- Checked for consistency in formatting and style across all updated docstrings.
- Reviewed and revised documentation for coherence and completeness.

### CODE REVIEW

File: src/caustics/lenses/pseudo_jaffe.py [hunk 1]
```diff
@@ -214,17 +280,34 @@ def reduced_deflection_angle(
...
             Dynamic parameter container.
 
         Returns
         --------
-        Tuple[Tensor, Tensor]
-            The deflection angle in the x and y directions.
+        x_component: Tensor
+            x-component of the deflection angle.
+
+            *Unit: radians*
```
Comment (ConnorStoneAstro) [inline src/caustics/lenses/pseudo_jaffe.py:304]: arcsec

File: src/caustics/lenses/pseudo_jaffe.py [hunk 2]
```diff
@@ -314,17 +409,29 @@ def convergence(
...
+
+        params: Packed, optional
             Dynamic parameter container.
 
         Returns
         -------
         Tensor
             The projected mass density.
+
+            *Unit: solMass/Mpc^2*
```
Comment (ConnorStoneAstro) [inline src/caustics/lenses/pseudo_jaffe.py:433]: unitless

File: src/caustics/lenses/sie.py [hunk 3]
```diff
@@ -114,17 +148,34 @@ def reduced_deflection_angle(
...
             Dynamic parameter container.
 
         Returns
         --------
-        Tuple[Tensor, Tensor]
-            The deflection angle in the x and y directions.
+        x_component: Tensor
+            The x-component of the deflection angle.
+
+            *Unit: radians*
```
Comment (ConnorStoneAstro) [inline src/caustics/lenses/sie.py:172]: arcsec

File: src/caustics/lenses/sie.py [hunk 4]
```diff
@@ -114,17 +148,34 @@ def reduced_deflection_angle(
...
+
+        y_component: Tensor
+            The y-component of the deflection angle.
+
+            *Unit: radians*
```
Comment (ConnorStoneAstro) [inline src/caustics/lenses/sie.py:177]: arcsec

File: src/caustics/lenses/sie.py [hunk 5]
```diff
@@ -197,17 +260,29 @@ def convergence(
...
+        params: Packed, optional
             Dynamic parameter container.
 
         Returns
         -------
         Tensor
-            The projected mass.
+            The projected mass density.
+
+            *Unit: solMass/megaparsec^2*
```
Comment (ConnorStoneAstro) [inline src/caustics/lenses/sie.py:284]: unitless

File: src/caustics/lenses/singleplane.py [hunk 6]
```diff
@@ -60,17 +63,34 @@ def reduced_deflection_angle(
...
             Dynamic parameter container.
 
         Returns
         -------
-        Tuple[Tensor, Tensor]
-            The total deflection angle in the x and y directions.
+        x_component: Tensor
+            The x-component of the deflection angle.
+
+            *Unit: radians*
```
Comment (ConnorStoneAstro) [inline src/caustics/lenses/singleplane.py:87]: arcsec

File: src/caustics/lenses/singleplane.py [hunk 7]
```diff
@@ -60,17 +63,34 @@ def reduced_deflection_angle(
...
+
+        y_component: Tensor
+            The y-component of the deflection angle.
+
+            *Unit: radians*
```
Comment (ConnorStoneAstro) [inline src/caustics/lenses/singleplane.py:92]: arcsec

File: src/caustics/lenses/singleplane.py [hunk 8]
```diff
@@ -98,17 +118,29 @@ def convergence(
...
+
+        params: Packed, optional
             Dynamic parameter container.
 
         Returns
         -------
         Tensor
             The total projected mass density.
+
+            *Unit: solMass/megaparsec^2*
```
Comment (ConnorStoneAstro) [inline src/caustics/lenses/singleplane.py:142]: unitless

File: src/caustics/lenses/sis.py [hunk 9]
```diff
@@ -21,17 +21,35 @@ class SIS(ThinLens):
...
+
+    th_ein (Optional[Union[Tensor, float]])
+        The Einstein radius of the lens.
+
+        *Unit: arcsec*
+
     s: float
         A smoothing factor, default is 0.0.
+
+        *Unit: meters*
```
Comment (ConnorStoneAstro) [inline src/caustics/lenses/sis.py:51]: arcsec

File: src/caustics/lenses/sis.py [hunk 10]
```diff
@@ -81,17 +99,34 @@ def reduced_deflection_angle(
...
             Dynamic parameter container.
 
         Returns
         -------
-        Tuple[Tensor, Tensor]
-            The deflection angle in the x and y directions.
+        x_component: Tensor
+            Deflection Angle
+
+            *Unit: radians*
```
Comment (ConnorStoneAstro) [inline src/caustics/lenses/sis.py:123]: arcsec

File: src/caustics/lenses/sis.py [hunk 11]
```diff
@@ -81,17 +99,34 @@ def reduced_deflection_angle(
...
+
+        y_component: Tensor
+            Deflection Angle
+
+            *Unit: radians*
```
Comment (ConnorStoneAstro) [inline src/caustics/lenses/sis.py:128]: arcsec

File: src/caustics/lenses/sis.py [hunk 12]
```diff
@@ -157,17 +204,29 @@ def convergence(
...
+
         params: (Packed, optional)
             Dynamic parameter container.
 
         Returns
         -------
         Tensor
             The projected mass density.
+
+            *Unit: solMass/megaparsec^2*
```
Comment (ConnorStoneAstro) [inline src/caustics/lenses/sis.py:228]: unitless

File: src/caustics/lenses/tnfw.py [hunk 13]
```diff
@@ -46,29 +46,55 @@ class TNFW(ThinLens):
...
-        Truncation scale. Ratio of truncation radius to scale radius (rt/rs).
+        Truncation scale. Ratio of truncation radius to scale radius.
+
+        *Unit: unitless*
+
     s: float
         Softening parameter to avoid singularities at the center of the lens.
         Default is 0.0.
+
+        *Unit: unitless*
```
Comment (ConnorStoneAstro) [inline src/caustics/lenses/tnfw.py:88]: arcsec

File: src/caustics/lenses/tnfw.py [hunk 14]
```diff
@@ -46,29 +46,55 @@ class TNFW(ThinLens):
...
+
     interpret_m_total_mass: boolean
         Indicates how to interpret the mass variable "m". If true
         the mass is interpreted as the total mass of the halo (good because it makes sense). If
         false it is interpreted as what the mass would have been within R200 of a an NFW that
         isn't truncated (good because it is easily compared with an NFW).
+
+        *Unit: unitless*
```
Comment (ConnorStoneAstro) [inline src/caustics/lenses/tnfw.py:96]: no units needed

File: src/caustics/lenses/tnfw.py [hunk 15]
```diff
@@ -167,23 +193,42 @@ def get_concentration(
...
+
         scale_radius: Optional[Tensor]
-            Scale radius of the TNFW lens (arcsec).
+            Scale radius of the TNFW lens.
+
+            *Unit: arcsec*
+
         tau: Optional[Tensor]
-            Truncation scale. Ratio of truncation radius to scale radius (rt/rs).
+            Truncation scale. Ratio of truncation radius to scale radius.
```
Comment (ConnorStoneAstro) [inline src/caustics/lenses/tnfw.py:220]: unitless

File: src/caustics/lenses/tnfw.py [hunk 16]
```diff
@@ -394,29 +522,50 @@ def mass_enclosed_2d(
...
+
         params: dict
             Dynamic parameter container.
 
         Returns
         -------
         Tensor
             Integrated mass projected in infinite cylinder within radius r.
+
+            *Unit: arcsec*
```
Comment (ConnorStoneAstro) [inline src/caustics/lenses/tnfw.py:567]: Msol

File: src/caustics/lenses/tnfw.py [hunk 17]
```diff
@@ -455,23 +604,48 @@ def physical_deflection_angle(
...
             Dynamic parameter container.
 
         Returns
         --------
-        tuple[Tensor, Tensor]
-            The physical deflection angles in the x and y directions (arcsec).
+        x_component: Tensor
+            Deflection Angle in x-direction.
+
+            *Unit: radians*
```
Comment (ConnorStoneAstro) [inline src/caustics/lenses/tnfw.py:643]: arcsec

File: src/caustics/lenses/tnfw.py [hunk 18]
```diff
@@ -455,23 +604,48 @@ def physical_deflection_angle(
...
+
+        y_component: Tensor
+            Deflection Angle in y-direction.
+
+            *Unit: radians*
```
Comment (ConnorStoneAstro) [inline src/caustics/lenses/tnfw.py:648]: arcsec

File: src/caustics/lenses/utils.py [hunk 19]
```diff
@@ -22,11 +22,19 @@ def get_pix_jacobian(
...
+        *Unit: arcsec*
+
     z_s: Tensor
         The redshift of the source.
 
+        *Unit: unitless*
+
     Returns
     --------
     The Jacobian matrix of the image position with respect
```
Comment (ConnorStoneAstro) [inline src/caustics/lenses/utils.py:40]: unitless

File: src/caustics/light/pixelated.py [hunk 20]
```diff
@@ -113,6 +142,9 @@ def brightness(
             The brightness of the source at the given coordinate(s).
             The brightness is determined by interpolating values
             from the source image.
+
+            *Unit: unitless*
```
Comment (ConnorStoneAstro) [inline src/caustics/light/pixelated.py:146]: flux

File: src/caustics/light/sersic.py [hunk 21]
```diff
@@ -25,22 +25,49 @@ class Sersic(Source):
...
+
     Re: Optional[Tensor]
         The scale length of the Sersic source.
+
+        *Unit: arcsec*
+
     Ie: Optional[Tensor]
         The intensity at the effective radius.
+
+        *Unit: unitless*
```
Comment (ConnorStoneAstro) [inline src/caustics/light/sersic.py:59]: flux

File: src/caustics/light/pixelated.py [hunk 22]
```diff
@@ -26,14 +26,25 @@ class Pixelated(Source):
...
+       *Unit: arcsec*
+
     y0 : Tensor, optional
         The y-coordinate of the source image's center.
+
+        *Unit: arcsec*
+
     image : Tensor, optional
         The source image from which brightness values will be interpolated.
+
```
Comment (ConnorStoneAstro) [inline src/caustics/light/pixelated.py:39]: flux

File: src/caustics/light/sersic.py [hunk 23]
```diff
@@ -25,22 +25,49 @@ class Sersic(Source):
...
+
     s: float
         A small constant for numerical stability.
+
+        *Unit: unitless*
```
Comment (ConnorStoneAstro) [inline src/caustics/light/sersic.py:64]: arcsec

File: src/caustics/light/sersic.py [hunk 24]
```diff
@@ -25,22 +25,49 @@ class Sersic(Source):
...
+
     lenstronomy_k_mode: bool
         A flag indicating whether to use lenstronomy to compute the value of k.
+
+        *Unit: unitless*
```
Comment (ConnorStoneAstro) [inline src/caustics/light/sersic.py:69]: no units needed

File: src/caustics/light/sersic.py [hunk 25]
```diff
@@ -63,24 +90,49 @@ def __init__(
...
+
+            *Unit: unitless*
+
         Re: Optional[Tensor]
             The scale length of the Sersic source.
+
+            *Unit: arcsec*
+
         Ie: Optional[Tensor]
             The intensity at the effective radius.
```
Comment (ConnorStoneAstro) [inline src/caustics/light/sersic.py:125]: flux

File: src/caustics/light/sersic.py [hunk 26]
```diff
@@ -63,24 +90,49 @@ def __init__(
...
         s: float
             A small constant for numerical stability.
+
+            *Unit: unitless*
+
         use_lenstronomy_k: bool
             A flag indicating whether to use lenstronomy to compute the value of k.
+
+            *Unit: unitless*
```
Comment (ConnorStoneAstro) [inline src/caustics/light/sersic.py:134]: no unit needed

File: src/caustics/light/sersic.py [hunk 27]
```diff
@@ -63,24 +90,49 @@ def __init__(
...
             The scale length of the Sersic source.
+
+            *Unit: arcsec*
+
         Ie: Optional[Tensor]
             The intensity at the effective radius.
         s: float
             A small constant for numerical stability.
+
+            *Unit: unitless*
```
Comment (ConnorStoneAstro) [inline src/caustics/light/sersic.py:129]: arcsec

File: src/caustics/light/sersic.py [hunk 28]
```diff
@@ -131,6 +189,8 @@ def brightness(
             The brightness of the source at the given point(s).
             The output tensor has the same shape as `x` and `y`.
 
+            *Unit: unitless*
```
Comment (ConnorStoneAstro) [inline src/caustics/light/sersic.py:192]: flux

**Review by ConnorStoneAstro — CHANGES_REQUESTED** [review]
Great job @uwcdc ! I flagged the units that need to be changed, I hope this is a good format for making those changes?

Some global changes, the general convention in astronomy for solar mass is to use `Msun` and for megaparsecs the unit is written as `Mpc`. It is a bit confusing that the `M` means different things in each one, but that is what everyone will be expecting.

Edit: Sorry, we checked what astropy does and it looks like `Msun` is actually the dominant convention.

**Review by ConnorStoneAstro — APPROVED** [review]
This looks amazing! Great stuff :)

**Comment by ConnorStoneAstro** [comment]
> This looks amazing! Great stuff :)

oh I see there's still a few more to do. But really good progress!

**Review by ConnorStoneAstro — CHANGES_REQUESTED** [review]
This is looking good, but there are still a bunch of unresolved comments for units that need to change

File: src/caustics/lenses/nfw.py [hunk 29]
```diff
@@ -148,17 +193,29 @@ def get_scale_radius(
...
             Mass of the lens.
+
+            *Unit: solMass*
+
         c: Tensor
             Concentration parameter of the lens.
+
+            *Unit: unitless*
+
         x: dict
```
[the remark this answers is not in the document]
Comment (uwcdc) [reply src/caustics/lenses/nfw.py:209]: Since it isn't related to the docstring, we will ignore for now.

File: src/caustics/lenses/tnfw.py (hunk 16 above)
Comment (lsetiawan) [reply src/caustics/lenses/tnfw.py]: This has been changed to `Msun` not the suggested Msol per discussions.

**Comment by lsetiawan** [comment]
@uwcdc and I have gone through all of the comments and resolved them one by one as they were outdated and now in the actual files. I think this PR is ready to go. Please approve when you agree. Thanks!

**Review by lsetiawan — APPROVED** [review]

**Review by ConnorStoneAstro — APPROVED** [review]
Awesome! I'm very excited about this update! :)

**Pull request merged**